pkg/controllers/ttl_after_finished_test.go

```diff
@@ -0,0 +1,189 @@
+		},
+		{
+			name:          "jobset running - TTL set",
+			jobset:        testutils.MakeJobSet(jobSetName, ns).TTLSecondsAfterFinished(10).Obj(),
+			clock:         clocktesting.NewFakeClock(now.Time),
+			expectExpired: false,
+			expectErr:     true,
+			expectErrStr:  "unable to find the status of the finished JobSet default/test-jobset",
+		},
+		{
+			name:          "jobset running - TTL not set",
+			jobset:        testutils.MakeJobSet(jobSetName, ns).TTLSecondsAfterFinished(10).Obj(),
+			clock:         clocktesting.NewFakeClock(now.Time),
+			expectExpired: false,
+			expectErr:     true,
+			expectErrStr:  "unable to find the status of the finished JobSet default/test-jobset",
+		},
+		{
+			name:          "jobset completed - deletion in progress",
+			jobset:        testutils.MakeJobSet(jobSetName, ns).TTLSecondsAfterFinished(10).CompletedCondition(now).DeletionTimestamp(&now).Obj(),
+			clock:         clocktesting.NewFakeClock(now.Time),
+			expectExpired: false,
+			expectErr:     false,
+		},
+		{
+			name:          "invalid jobset completion time",
+			jobset:        testutils.MakeJobSet(jobSetName, ns).TTLSecondsAfterFinished(10).CompletedCondition(metav1.Time{}).Obj(),
+			clock:         clocktesting.NewFakeClock(now.Time),
+			expectExpired: false,
+			expectErr:     true,
+			expectErrStr:  "unable to find the time when the JobSet default/test-jobset finished",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			r := &JobSetReconciler{clock: tc.clock}
+			expired, err := checkIfTTLExpired(ctx, r.clock, tc.jobset)
+			if tc.expectErr != (err != nil) {
+				t.Fatalf("expected error: %v, got: %v", tc.expectErr, err)
+			}
+			if tc.expectErr && !strings.Contains(err.Error(), tc.expectErrStr) {
+				t.Errorf("expected error string '%s' to be in '%v'", tc.expectErrStr, err)
+			}
+			if expired != tc.expectExpired {
+				t.Errorf("expected expired to be %v, got %v", tc.expectExpired, expired)
+			}
+		})
+	}
+}
+
+func TestTimeLeft(t *testing.T) {
+	var (
+		jobSetName = "test-jobset"
+		ns         = "default"
+	)
+
+	now := metav1.Now()
+
+	tests := []struct {
+		name             string
+		jobset           *jobset.JobSet
+		since            *time.Time
+		expectErr        bool
+		expectErrStr     string
+		expectedTimeLeft *time.Duration
+	}{
+		{
+			name:         "jobset not finished",
+			jobset:       testutils.MakeJobSet(jobSetName, ns).Obj(),
+			expectErr:    true,
+			expectErrStr: "unable to find the status of the finished JobSet default/test-jobset",
+		},
+		{
+			name:   "jobset completed now, nil since",
+			jobset: testutils.MakeJobSet(jobSetName, ns).TTLSecondsAfterFinished(0).CompletedCondition(now).Obj(),
+			since:  nil,
+		},
+		{
+			name:             "jobset completed now, 0s TTL",
+			jobset:           testutils.MakeJobSet(jobSetName, ns).TTLSecondsAfterFinished(0).CompletedCondition(now).Obj(),
+			since:            &now.Time,
```

Comment:
In the `timeLeft()` function being tested here, we changed the `since` parameter to be called `now` (since the purpose of the variable is to hold the current time). We should make the corresponding change here, and call this `now` or `currTime`.